java/client/src/org/openqa/selenium/remote/internal/HttpClientFactory.java

```diff
@@ -114,18 +119,18 @@ public HttpClient getGridHttpClient(int connection_timeout, int socket_timeout)
         .build();
   }
 
-  private SocketConfig createSocketConfig() {
+  private SocketConfig createSocketConfig(int socketTimeout) {
     return SocketConfig.custom()
         .setSoReuseAddress(true)
-        .setSoTimeout(TIMEOUT_THREE_HOURS)
+        .setSoTimeout(socketTimeout > 0 ? socketTimeout : TIMEOUT_THREE_HOURS)
```

Comment:
Okay.  I guess the fact I was confused illustrates the problem I was hoping to address in any event :-)  In that case, throwing an exception about an invalid timeout value seems more reasonable to me than silently ignoring it with a substituted default value.